R/module_plots.R

```diff
@@ -21,21 +21,29 @@ plotsServer = function(id, cluster_choice) {
   shiny::moduleServer(id, function(input, output, session) {
     ns = session$ns
 
+    # disable dropdown initially
+    shiny::observe({
+      shinyjs::disable("plot_type")
+    })
+
     # all available plots
     all_files = shiny::reactive({
       return(get_all_files(cluster_choice()))
     }) %>%
       shiny::bindCache(cluster_choice())
 
     # drop down for plots
-    output$choose_plot = shiny::renderUI({
+    shiny::observeEvent(all_files(), {
       all_images = filter_by_filetype(filenames = all_files(),
-                         filetypes = c("png", "PNG"))
-      shiny::selectInput(ns("plot_type"),
-                         label = "Select plot type:",
-                         choices = all_images)
-    }) %>%
-      shiny::bindCache(cluster_choice())
+                                      filetypes = c("png", "PNG"))
+      if (length(all_images) != 0) {
+        shinyjs::enable("plot_type")
+      }
```

Comment:
I'm wondering if there's a small bug here, but couldn't find an example to prove it.

- User opens Tree Logistic Growth Rate
- User expands "Download Files" and opens "Plots" subpanel
- User selects a node that does have .png figures associated with it
- [Select-plot-type dropdown is activated]
- User selects a node that does not have a .png figures associated with it
- [Select-plot-type dropdown should be inactivated, but I don't think it will be]

Could you add `else {shinyjs::disable("plot_type")}`

Reply:
I believe the issue will affect the rdsServer and tablesServer as well.

Reply:
Yes, you're right - I found an example where this is happening. I've added in the `else` statements and that seems to have fixed it. 

I'll also add an issue that we may want to move this to a separate module/function in the longer term.